src/components/Identicon/index.tsx

```diff
@@ -1,64 +1,60 @@
 import * as React from 'react'
-import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
 
-import { toDataUrl } from './blockies'
+import makeBlockie from 'ethereum-blockies-base64'
+import { useEffect, useState } from 'react'
 
-export default class Identicon extends React.PureComponent<any> {
-  private identicon: React.RefObject<HTMLImageElement>
+type Props = {
+  address: string
+  className?: string
+  diameter: number
+}
 
-  static defaultProps = {
-    className: '',
-  }
+type StyleProps = {
+  width: number
+  height: number
+}
 
-  constructor(props) {
-    super(props)
+const getStyleFrom = (diameter: number): StyleProps => ({
+  width: diameter,
+  height: diameter,
+})
 
-    this.identicon = React.createRef<HTMLImageElement>()
-  }
+const generateBlockieIdenticon = (address: string, diameter: number): HTMLImageElement => {
+  const image = new window.Image()
+  image.src = makeBlockie(address)
+  image.height = diameter
+  image.width = diameter
+  image.style.borderRadius = `${diameter / 2}px`
 
-  componentDidMount = () => {
-    const { address, diameter } = this.props
-    const image = this.generateBlockieIdenticon(address, diameter)
-    if (this.identicon.current) {
-      this.identicon.current.appendChild(image)
-    }
-  }
+  return image
+}
+
+const Identicon = (props: Props) => {
+  const { address, diameter, className } = props
+  const style: StyleProps = getStyleFrom(diameter)
+
+  const [identicon, setIdenticon] = useState(null)
 
-  componentDidUpdate = () => {
-    const { address, diameter } = this.props
-    const image = this.generateBlockieIdenticon(address, diameter)
+  useEffect(() => {
```

Comment:
do we really need this complex logic with multiple `useEffect`?

what happens if you just do
```
const iconSrc = React.useMemo(() => makeBlockie(address), [address])
```

Also, the types are wrong:
```
React.createRef<HTMLImageElement>
```
used:
```
<div className={className} ref={identicon} style={style} />
```

Reply:
Good idea! please check this out again, it's simpler now

Reply:
Looks good, thank you :)

One thing, can we make use of styled components to stay consistent and remove `getStyleProps`?

something like this:
```
const Image = styled.img<{ size: number }>`
  width: ${({ size }) => `${size}px`};
  height: ${({ size }) => `${size}px`};
`

const Identicon: React.FC<Props> = ({ size = 20, address }) => <Image ... />
```

Reply:
Done :)